internal/dcache/rpc/server/handler.go

```diff
@@ -46,44 +56,861 @@ var _ service.ChunkService = &ChunkServiceHandler{}
 
 // ChunkServiceHandler struct implements the ChunkService interface
 type ChunkServiceHandler struct {
-	locks *common.LockMap
-	// more fields can be added here as needed
+	locks   *common.LockMap
+	fsIDMap map[string]*rvInfo // map to store the fsID to rvInfo mapping
+}
+
+type rvInfo struct {
+	rvID     string       // fsID for this RV
+	rvName   string       // rv0, rv1, etc.
+	cacheDir string       // cache dir path for this RV
+	mvMap    sync.Map     // sync map of MV id against mvInfo
+	mvCount  atomic.Int64 // count of MVs for this RV, this should be updated whenever a MV is added or removed from the sync map
+}
+
+type mvInfo struct {
+	mu           sync.Mutex
+	mvName       string       // mv0, mv1, etc.
+	componentRVs []string     // sorted list of component RVs for this MV
+	chunkOpsCnt  atomic.Int64 // count of chunk operations (get, put or remove) for this MV
+	blockOpsFlag atomic.Bool  // block chunk operations (get, put or remove) for this MV. This is done when the .sync dir contents are being moved to the regular MV dir
+	syncInfo                  // sync info for this MV
+}
+
+type syncInfo struct {
+	isSyncing bool   // is the MV in syncing state
```

Comment:
I'd encourage you to be generous about writing good descriptive comments.
f.e.

```
	isSyncing bool   // Is the MV in syncing state?
	                 // An MV enters syncing state after StartSync command is successfully executed.
	                 // In syncing state, PutChunk requests corresponding to client writes will be
	                 // saved in the mv#.sync folder. Similarly a successful EndSync will take an
	                 // MV out of syncing state.
	syncID    string // sync ID for this MV if it is in syncing state.
	                 // This is returned in the StartSync response and EndSync should carry this.
```

Reply:
Done